### ISSUE TITLE
Suggestion in how to train the network

### ISSUE THREAD

**Issue by akineeic** [swaroop1904/keypointnet#6]
Thanks for your wonderful sharing first! Recently I'm trying to train keypointnet using the [code in tensorflow](https://github.com/tensorflow/models/tree/v1.13.0/research/keypointnet). But the output predicted images are terrible. And I saw the result showed in your presentation is much more better than mine. So I want to know do your have any suggestion in how to train the network especially the value of the parameters like `batch_size`, `learning_rate`, the weight of each loss, etc. Any suggestion will be welcomed sincerely. Thanks for your attention.

**Comment by KAOeric** [comment]
Hello, mr.akineeic did you already solve your probelms? I have the same problem like yours, so could you give me some suggestion, Thanks!

**Comment by swaroop1904** [comment]
Sorry for the delay in replying. Batch size: 16, learning rate: 1e-4, number of steps: 700k for plane and 600k for car. For the losses, we used the same weights as mentioned in the paper. The weights of the losses can be varied depending on the what is the issue. In case the keypoints are outside the outside, then weightage of the silhouette loss can be increased to make sure that all points are inside the object, if the keypoints are concentrated at a specific location, then the variance of the noise added can be increased and also weightage of separation loss can be increased.

**Comment by KAOeric** [comment]
Thanks for your answer. But my computer only can run batch size:4 ,how can i adjust the learning rate. Could you share the output of predicted images that you run or how the code you successful to run. Thanks!

**Comment by KAOeric** [comment]
Mr.Swaroop, In addition, I don’t know if I can have your e-mail address. I want to show you some of my achievements. I would like to ask you to help me solve my problem.

**Comment by akineeic** [comment]
> Hello, mr.akineeic did you already solve your probelms? I have the same problem like yours, so could you give me some suggestion, Thanks!

Sorry for the late reply, I haven't solve this problem yet. Maybe you can try the suggestion of the author above.

**Comment by swaroop1904** [comment]
> Thanks for your answer. But my computer only can run batch size:4 ,how can i adjust the learning rate. Could you share the output of predicted images that you run or how the code you successful to run. Thanks!

The outputs for the images are as shown in the readme. You could monitor individual losses to check if there is any specific loss which is not reducing. As for learning rate, you will need to specifically tune it for the batch size.

**Comment by akineeic** [comment]
@swaroop1904 Sorry to bother you again, I want to know if you have measured the angular error of your trained model(the indicators mentioned in the paper). Could you meet the level in the paper?

**Comment by swaroop1904** [comment]
I haven't explicitly created the histogram for the various angular errors for the trained model. But I had explicitly monitored the angular loss to make sure that the network is optimizing to reduce the angular error.

**Comment by akineeic** [comment]
Got it. Thanks for your quick reply!
